backend/substrapp/compute_tasks/image_builder.py

```diff
@@ -42,17 +42,16 @@ def wait_for_image_built(function: orchestrator.Function, channel: str) -> None:
 def load_remote_function_image(function: orchestrator.Function, channel: str) -> None:
     container_image_tag = utils.container_image_tag_from_function(function)
     # Ask the backend owner of the function if it's available
-    logger.info(
-        f"Initial function URI {function.function_address.uri}; "
-        f"modified URI{function.function_address.uri.replace('file', 'image')}"
-    )
+
+    # Necessary to query the DB again to get an updated Function object
+    # Else the image URI is not available
+    api_function = ApiFunction.objects.get(key=function.key)
 
     function_image_content = organization_client.get(
         channel=channel,
         organization_id=function.owner,
-        # TODO create a clean Address for function image
-        url=function.function_address.uri.replace("file", "image"),
-        checksum=None,
+        url=api_function.image_address,
+        checksum=api_function.image_checksum,
```

Comment:
Could we use `client.query_function(function.key)` instead of `ApiFunction.objects.get(key=function.key)` to avoid creating more dependencies between worker and DB?

Reply:
What client are you refering to ? Because from what I understand, this is to force the `organization_client` to get the updated value of the function.